site/src/xServices/workspaces/workspacesXService.ts

```diff
@@ -1,64 +1,263 @@
+            invoke: {
+              id: "getWorkspace",
+              src: "getWorkspace",
+              onDone: [
+                {
+                  target: "waitingToBeUpdated",
+                  cond: "isOutdated",
+                  actions: ["assignUpdatedData"],
+                },
+                {
+                  target: "idle",
+                  actions: ["assignUpdatedData", "displayUpdatedSuccessMessage"],
+                },
+              ],
+            },
+          },
+        },
+      },
+    },
+  },
+  {
+    guards: {
+      isOutdated: (_, event) => event.data.outdated,
+    },
+    services: {
+      getTemplate: (context) => API.getTemplate(context.data.template_id),
+      startWorkspace: (context) => {
+        if (!context.updatedTemplate) {
+          throw new Error("Updated template is not loaded.")
+        }
+
+        return API.startWorkspace(context.data.id, context.updatedTemplate.active_version_id)
+      },
+      getWorkspace: (context) => API.getWorkspace(context.data.id),
+    },
+    actions: {
+      assignUpdatedTemplate: assign({
+        updatedTemplate: (_, event) => event.data,
+      }),
+      assignLatestBuild: assign({
+        data: (context, event) => {
+          return {
+            ...context.data,
+            latest_build: event.data,
+          }
+        },
+      }),
+      displayUpdateVersionError: (_, event) => {
+        const message = getErrorMessage(event.data, "Error on update workspace version.")
+        displayError(message)
+      },
+      displayUpdatingVersionMessage: () => {
+        displayMsg("Updating workspace...")
+      },
+      assignQueuedStatus: assign({
+        data: (ctx) => {
+          return {
+            ...ctx.data,
+            latest_build: {
+              ...ctx.data.latest_build,
+              job: {
+                ...ctx.data.latest_build.job,
+                status: "pending" as TypesGen.ProvisionerJobStatus,
+              },
+            },
+          }
+        },
+      }),
+      displayUpdatedSuccessMessage: () => {
+        displaySuccess("Workspace updated successfully.")
+      },
+      assignUpdatedData: assign({
+        data: (_, event) => event.data,
+      }),
+    },
+  },
+)
+
+/**
+ * Workspaces machine
+ *
+ * It is used to control the state of the workspace list
+ **/
+
+export type WorkspaceItemMachineRef = ActorRefFrom<typeof workspaceItemMachine>
+
+interface WorkspacesContext {
+  workspaceRefs?: WorkspaceItemMachineRef[]
   filter?: string
   getWorkspacesError?: Error | unknown
 }
 
-type WorkspaceEvent = { type: "GET_WORKSPACE"; workspaceId: string } | { type: "SET_FILTER"; query: string }
+type WorkspacesEvent = { type: "GET_WORKSPACES"; query: string } | { type: "UPDATE_VERSION"; workspaceId: string }
 
 export const workspacesMachine = createMachine(
   {
-    tsTypes: {} as import("./workspacesXService.typegen").Typegen0,
+    tsTypes: {} as import("./workspacesXService.typegen").Typegen1,
```

Comment:
What's this `Typegen1` vs `Typegen0`?

Reply:
It is auto-generated by XState types: https://xstate.js.org/docs/guides/typescript.html#typegen